src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/JsonSourceGeneratorHelper.cs

```diff
@@ -0,0 +1,559 @@
+                {
+                    Diagnostics.Add(Diagnostic.Create(_notSupported, Location.None, new string[] { root.Name, property.PropertyType.Name }));
+                    return false;
+                }
+                foreach (Type handlingType in GetTypesToGenerate(property.PropertyType))
+                {
+                    GenerateForMembers(root, handlingType, seenTypes, ref isSuccessful);
+                }
+            }
+
+            // Try to generate current type info now that fields and property types have been resolved.
+            isSuccessful &= AddImportsToTypeClass(type, properties, fields, source);
+            isSuccessful &= InitializeContextClass(type, className, source);
+            isSuccessful &= InitializeTypeClass(className, source);
+            isSuccessful &= TypeInfoGetterSetter(type, source);
+            isSuccessful &= InitializeTypeInfoProperties(properties, source);
+            isSuccessful &= GenerateTypeInfoConstructor(type, className, properties, fields, source);
+            isSuccessful &= GenerateCreateObject(type, source);
+            isSuccessful &= GenerateSerialize(type, properties, source);
+            isSuccessful &= GenerateDeserialize(type, properties, source);
+            isSuccessful &= FinalizeTypeAndContextClasses(source);
+
+            if (isSuccessful)
+            {
+                Diagnostics.Add(Diagnostic.Create(_generatedTypeClass, Location.None, new string[] { root.Name, className }));
+
+                // Add generated typeinfo for current traversal.
+                Types.Add(type, source.ToString());
+            }
+            else
+            {
+                Diagnostics.Add(Diagnostic.Create(_failedToGenerateTypeClass, Location.None, new string[] { root.Name, className }));
+
+                // If not successful remove it from found types hashset and add to failed types list.
+                seenTypes.Remove(type);
+                _failedTypes.Add(type);
+            }
+
+            return isSuccessful;
+        }
+
+        // Call recursive type generation if unseen type and check for success and cycles.
+        void GenerateForMembers(Type root, Type currentType, HashSet<Type> seenTypes, ref bool isSuccessful)
+        {
+            // If new type, recurse.
+            if (IsNewType(currentType, seenTypes))
+            {
+                bool wasSuccessful = GenerateClassInfo(root, seenTypes, currentType.Name, currentType);
+                isSuccessful &= wasSuccessful;
+
+                if (!wasSuccessful)
+                {
+                    Diagnostics.Add(Diagnostic.Create(_failedToAddNewTypesFromMembers, Location.None, new string[] { root.Name, currentType.Name }));
+                }
+            }
+        }
+
+        // Check if current type is supported to be iterated over.
+        private bool IsSupportedType(Type type)
+        {
+            if (type is TypeWrapper typeWrapper)
```

Comment:
only time type is not a typewrapper is when it is a primitive supported type. we return true because converters along with typeclasses are already within default instance for jsoncontext.

Reply:
Changed to isMemberSuccessful since wasSuccessful was the recursed member generation